### PR TITLE
6

### PR DESCRIPTION
**Pull request by JAMESBRADLEY99**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -92,4 +92,55 @@ describe(' GET /api/reviews/:review_id', () => {
...
+            .expect(200)
+            .then((res) => {
+                expect(res.body.msg).toBe('no comments for review')
+            })
+    });
+
+    test('404s on non existent id', () => {
+        return request(app)
+            .get('/api/reviews/10000/comments')
+            .expect(404)
```
Comment (Slyose) [inline __tests__/api.test.js:153]: An error message should be accompanying this 404 error, test for that as well.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -92,4 +92,55 @@ describe(' GET /api/reviews/:review_id', () => {
...
+    });
 });
```
Comment (Slyose) [inline __tests__/api.test.js:158]: You need another test for error 400's. That's when the id is an incorrect data type (not a number)

Think about how you can take advantage of built-in SQL error handling to achieve this

File: __tests__/api.test.js [hunk 3]
```diff
@@ -92,4 +92,55 @@ describe(' GET /api/reviews/:review_id', () => {
             .get('/api/reviews/1000')
             .expect(404)
     });
+});
+
+describe(' 6. GET /api/reviews/:review_id/comments', () => {
+    test('returns correct comments', () => {
+        return request(app)
+            .get('/api/reviews/2/comments')
+            .expect(200)
```
Comment (Slyose) [inline __tests__/api.test.js:110]: Another test is needed for the status code 200 OK - you need a test where the review has no comments so an empty array is returned.

File: Models/models.js [hunk 4]
```diff
@@ -30,4 +30,22 @@ exports.selectReviewById = (review_id) => {
         }
         return review.rows[0]
     })
+}
+
+exports.selectCommentsByReviewId = (review_id) => {
+    return db.query(
+        `SELECT * FROM reviews WHERE review_id = $1`, [review_id]
```
Comment (Slyose) [inline Models/models.js:37]: "comments should be served with the most recent comments first"
What SQL method can you use here to achieve this?

**Review by Slyose — CHANGES_REQUESTED** [review]
Needs some tweaks before it's ready to be merged
